3di | Input needed for 'About' tools articles (#6974)
ros2/ros2_documentation: source/About-builds.rst

Builds
======

The ability to work with multiple packages together is useful for developing complex robotics systems.
This article summarizes the ROS developer tools and guidance available to help with builds.

**Area: builds, tools | Content-type: about | Experience: beginner, intermediate, expert**

.. contents:: Table of Contents
   :local:

Summary
-------

Most ROS systems use multiple packages working together.
Build tools in ROS enable you to work with multiple packages in a single project and to share your packages with other members of the ROS community.

Core ROS packages:

> * `colcon <https://docs.ros.org/en/lyrical/Tutorials/Beginner-Client-Libraries/Colcon-Tutorial.html>`_: Tool for creating and building a ROS workspace.

Reviewer: ```suggestion
* `colcon <https://docs.ros.org/en/lyrical/Tutorials/Beginner-Client-Libraries/Colcon-Tutorial.html>`_: A build tool for building packages in a ROS workspace.
```

> * `ament_cmake <https://docs.ros.org/en/lyrical/How-To-Guides/Ament-CMake-Documentation.html>`_: Build system for CMake-based packages in ROS.

Reviewer: ```suggestion
* `ament_cmake <https://docs.ros.org/en/lyrical/How-To-Guides/Ament-CMake-Documentation.html>`_: A CMake library to facilitate inter package dependencies in ROS.
```

* `rosdoc2 <https://docs.ros.org/en/rolling/How-To-Guides/Documenting-a-ROS-2-Package.html>`_: Tool for documenting a ROS package.

> QUESTION FOR THE REVIEWER: IN THE SPREADSHEET THERE IS A QUESTION MARK NEXT TO THIS ENTRY.

Reviewer: No rosdoc2 is a documentation tool not a build tool.

DO WE NEED TO INCLUDE ros2doc IN THIS LIST?

QUESTION FOR THE REVIEWER: ARE THERE ANY OTHER CORE ROS PACKAGES THAT WE CAN LIST HERE?

Community-contributed packages:

> QUESTION FOR THE REVIEWER: ARE THERE ANY COMMUNITY-CONTRIBUTED PACKAGES THAT WE CAN LIST HERE?

Reviewer: I would flag the extra one above as potentially community contribued inline versus a whole separate section.
